src/baseline_automatic_pipeline_system/coreference.py

```diff
@@ -15,14 +20,134 @@ def cluster_entity_mentions(mention_list):
+    :param lexical_wt: the lexical weight for the scoring function which compares two proposition clusters.
+     (the argument weight is 1- lexical_wt)
+    :param argument_match_ratio: the minimum argument alignment threshold between propositions
+    :return: a numeric value denoting the similarity between the mention to the cluster
+    """
+
+    from eval_predicate_coref import some_word_match
+
+    if prop[1] != 'IMPLICIT':
+        prop_terms = prop[1]
+        lexical_matches = 0
+        for other_prop in cluster:
+            if other_prop[1] != 'IMPLICIT':
+                if some_word_match(other_prop[1], prop_terms):
+                    lexical_matches += 1
+
+        lexical_score = lexical_matches / (1.0 * len(cluster))
+
+
+    else:
+        lexical_score = 0
+        lexical_wt = 0
+
+    argument_score = len([other for other in cluster if
+                            (some_arg_match(other[2], prop[2], argument_clusters, argument_match_ratio))]) / (
+                        1.0 * len(cluster))
+
+    return lexical_wt * lexical_score + (1 - lexical_wt) * argument_score
+
+
+def cluster_prop_mentions(mention_list, score, argument_clusters, lexical_wt, argument_match_ratio):
+    """
+    Cluster the predicate mentions in a greedy way: assign each predicate to the first
+    cluster with similarity score > 0.5. If no such cluster exists, start a new one.
+    :param mention_list: the mentions to clustert
+    :param score: the score function that receives a mention and a cluster and returns a score
+    :param argument_clusters: a clustering of entity-mentions to coreference chains.
+    :param lexical_wt: the lexical weight for the scoring function which compares two proposition clusters.
+     (the argument weight is 1- lexical_wt)
+    :param argument_match_ratio: the minimum argument alignment threshold between propositions
+    :return: clusters of mentions
+    """
+    clusters = []
+
+    for mention in mention_list:
+        found_cluster = False
+        for cluster in clusters:
+            cluster_mention_score = score(mention, cluster, argument_clusters, lexical_wt, argument_match_ratio)
+            if cluster_mention_score > 0.5:
+                cluster.append(mention)
+                found_cluster = True
+                break
+
+        if not found_cluster:
+            clusters.append([mention])
+
+    return clusters
+
+
+def some_arg_match(prop_mention1_info, prop_mention2_info, argument_clusters, arg_match_ratio):
+    """
+    Finds if at least some % of arguments of two propositions are coreferent.
+    :param prop_mention1_info: First proposition mention
+    :param prop_mention2_info: Second proposition mention
+    :param argument_clusters: The clusters of arguments obtained by coreference
+    :param arg_match_ratio: criteria of argument overlap = no of aligned arguments / no of arguments(minimum of proposition1 and proposition2)
```

Comment:
instead of "no of (aligned) arguments.." use "number of arguments" or "#arguments..."